packages/indy-vdr/src/dids/IndyVdrIndyDidRegistrar.ts

```diff
@@ -386,3 +563,32 @@ interface IndyVdrDidCreateOptionsWithoutDid extends IndyVdrDidCreateOptionsBase
 }
 
 export type IndyVdrDidCreateOptions = IndyVdrDidCreateOptionsWithDid | IndyVdrDidCreateOptionsWithoutDid
+
+type ParseInputOk = {
+  status: 'ok'
+  did: string
+  verificationKey: Key
+  namespaceIdentifier: string
+  namespace: string
+  submitterNamespaceIdentifier: string
+  seed: Buffer | undefined
+  privateKey: Buffer | undefined
+}
+
+type parseInputError = { status: 'error'; errorMessage: string }
+
+type ParseInputResult = ParseInputOk | parseInputError
+
+export interface EndorseDidTxAction extends DidOperationStateActionBase {
+  name: 'signNymTx'
+  did: string
+  submitterDid: string
+  nymRequest: NymRequest | CustomRequest
+  attribRequest?: AttribRequest | CustomRequest
+  didDocument: DidDocument
+  verificationKey: Key
+  role?: string
+  alias?: string
```

Comment:
Are all these fields required for the endorsement? It feels a bit bloated, and I'm thinking we could just skip some of the values.

Also, should we rename `submitterDid` to `endorserDid` here? That feels more aligned with the endorsement model.

From looking at the indy vdr endorser API, we just need: `nymRequest`, optional `attribRequest`, `endorserDid`. 

The other fields seems not needed? Is that because we want to continue the create action? Shouldn't those values be inputted from the other fields of the create action?